### PR TITLE
Add purge locks

### PR DESCRIPTION
**Pull request by tsmr**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | none

Add purge automatic locks (for inventory software)

### CODE REVIEW

File: front/lock.form.php [hunk 1]
```diff
@@ -58,7 +58,31 @@
          }
       }
    }
+
+   //Execute hook to unlock fields managed by a plugin, if needed
+   Plugin::doHookFunction('unlock_fields', $_POST);
+
+} else if (isset($_POST['itemtype']) && isset($_POST["purge"])) {
+   $itemtype    = $_POST['itemtype'];
```
Comment (trasher) [inline front/lock.form.php]: This is mostly a copy/paste of the code in the other if part, should be factorized.

Also, I wonder if the `$actions` list should not be defined in `inc/define.php`.

Comment (yllen) [inline front/lock.form.php]: closeArrowMassives and openArrowMassives are deprecated since 0.84 so must not be used

Comment (trasher) [inline front/lock.form.php]: I agree with that; but this is not the point of this PR. See https://github.com/glpi-project/glpi/issues/2394

**Comment by trasher** [comment]
@tsmr ping? Do you plan to update this PR or shall we close it?

**Comment by tsmr** [comment]
for conflict ?

**Comment by trasher** [comment]
Not only for the conflict; you didi not change anything nor answer since I've requsted changes (https://github.com/glpi-project/glpi/pull/2026#discussion_r113375873)

**Comment by tsmr** [comment]
I only add an action to the existant call.. I must change all massiveaction system on it ?

**Comment by trasher** [comment]
Huh... What? 

I ask you:
1- to factorize the code (most of the new code already exists);
2- to use - if relevant - a "global declaration" for the actions (would be useless once code has been factorized maybe -- I don't know if this is used elsewhere).

**Pull request merged**